File: apps/mobile-app/ios/VerseofTheDay/WidgetHelper.m
#import <Foundation/Foundation.h>
#import <WidgetKit/WidgetKit.h>
#import <React/RCTBridgeModule.h>

@interface WidgetHelper : NSObject <RCTBridgeModule>
@end

@implementation WidgetHelper

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(reloadWidget) {
  if (@available(iOS 14.0, *)) {
    [WidgetCenter.sharedWidgetCenter reloadTimelinesOfKind:@"widget"]; // Ensure "widget" matches your widget identifier
  } else {
    NSLog(@"Widget refresh is not supported on this iOS version.");
  }
}

@end

File: apps/mobile-app/ios/WidgetHelper.swift
//
//  WidgetHelper.swift
//  VerseofTheDay
//
//  Created by Aremu Oluwagbamila on 21/01/2025.
//

import Foundation
import WidgetKit

@objc(WidgetHelper)
class WidgetHelper: NSObject {
  
  // Method to reload the widget timeline
  @objc static func reloadWidget() {
    if #available(iOS 14.0, *) {
      // Reload the widget timeline for "MyWidget"
      WidgetCenter.shared.reloadTimelines(ofKind: "widget")
    } else {
      NSLog("WidgetKit is not available on this iOS version.")
    }
  }
}
